helm/fluent-logshipping-app/templates/configmap.yaml

```
@@ -392,9 +392,9 @@ data:
         total_file_size              {{ .Values.outputs.aws.S3.totalFileSize }}
         send_content_md5             true
         {{- if .Values.outputs.aws.S3.bucketPathPrefix }}
-        s3_key_format                {{ printf "%s/ssh%s" .Values.outputs.aws.S3.bucketPathPrefix .Values.outputs.aws.S3.s3_object_key_format }}
+        s3_key_format                {{ printf "%s/ssh%s" (regexReplaceAll "^/*" .Values.outputs.aws.S3.bucketPathPrefix "/") .Values.outputs.aws.S3.s3_object_key_format }}
```

I would feel more comfortable using a trim function instead of a regex.

```suggestion
        s3_key_format                {{ printf "%s/ssh%s" (trimPrefix "/" .Values.outputs.aws.S3.bucketPathPrefix) .Values.outputs.aws.S3.s3_object_key_format }}
```

---

This regex actually adds a leading slash when it's not present as this value requries a leading slash or will cause the pod to crash :)

---

Ok, my bad I misunderstood the code then. But still regex are last resort option to me.

I would suggest to do the following, trim any potential `/` prefix and then add a `/` prefix. This way we are sure there is always a `/` prefix.

---

We can but that's what this regex does already xD

---

but sure